src/xdocs/config_filters.xml

```diff
@@ -860,6 +855,60 @@ public class UserService {
 &lt;suppress-xpath checks=&quot;.*&quot; query=&quot;/CLASS_DEF[@text='TestClass']
           //METHOD_DEF[@text='testMethod']//VARIABLE_DEF[@text='testVariable']&quot;/&gt;
         </source>
+        <p>
+          In the following sample, violations for <code>LeftCurly</code> check will be suppressed
+          for classes with name <i>Main</i> or for methods with name <i>calculate</i>.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;LeftCurly&quot; query=&quot;/CLASS_DEF[@text='Main']/OBJBLOCK/LCURLY
+          | //METHOD_DEF[@text='calculate']/SLIST&quot;/&gt;
+        </source>
+        <p>
+          The following example demonstrates how to suppress <code>RequireThis</code> errors for
+          variable <i>age</i> inside <i>changeAge</i> method.
+        </p>
+        <source>
+&lt;suppress-xpath checks=&quot;RequireThis&quot; query=&quot;/CLASS_DEF[@text='InputTest']
+          //METHOD_DEF[@text='changeAge']//ASSIGN[@text='age']/IDENT&quot;/&gt;
+        </source>
+        <source>
+public class InputTest {
+  private int age = 23;
+
+  public void changeAge() {
+    age = 24; //violation will be suppressed
+  }
+}
+        </source>
+        <p>
+          Suppress <code>IllegalThrows</code> violations only for methods with name
+          <i>throwsMethod</i> and only for <code>RuntimeException</code> exceptions.
+          Double colon is used for axis iterations. In the following example <code>ancestor</code>
+          axis is used to iterate all ancestor nodes of the current node with type
```

Comment:
please provide some links where Xpath is explained (all axises, ... ) and all supported query features of it. Do not expect that users are know Xpath very well. Even I do not know it :).

Reply:
What do you think about [w3schools](https://www.w3schools.com/xml/xpath_axes.asp) link?

Reply:
looks good, lets start at least from it.

Reply:
done